### PR TITLE
make sub-ordinate dictionaries inherit injector from multiplexManager's dictionary.

### LINKED ISSUE #1379: MultiplexManager failing to propagate its dictionary's Injector to subordinate DataStores
ELIDE permits you to pass an Injector when creating an EntityDictionary. This can be used to populate entity fields when instances are created, (e.g. within Spring environment)

When used in the context of the MultiplexManager, this injector instance is *not* currently passed down to the sub-ordinate dictionaries, which the multiplex manager creates on behalf of its sub-ordinate datastores.

## Expected Behavior
When creating a sub-ordinate dictionary, the MultiplexManager should pass both both the security check mappings and injector from it's dictionary.
```java
 for (DataStore dataStore : dataStores) {
            EntityDictionary subordinateDictionary = new EntityDictionary(dictionary.getCheckMappings(), dictionary.getInjector());

            dataStore.populateEntityDictionary(subordinateDictionary);
            for (Class<?> cls : subordinateDictionary.getBindings()) {
                // route class to this database manager
                this.dataStoreMap.put(cls, dataStore);
                // bind to multiplex dictionary
                dictionary.bindEntity(cls);
                dictionary.bindInitializer(subordinateDictionary::initializeEntity, cls);
            }
        }
```

## Current Behavior
Currently only the security check mappings are passed down.
```java
 for (DataStore dataStore : dataStores) {
            EntityDictionary subordinateDictionary = new EntityDictionary(dictionary.getCheckMappings());

            dataStore.populateEntityDictionary(subordinateDictionary);
            for (Class<?> cls : subordinateDictionary.getBindings()) {
                // route class to this database manager
                this.dataStoreMap.put(cls, dataStore);
                // bind to multiplex dictionary
                dictionary.bindEntity(cls);
                dictionary.bindInitializer(subordinateDictionary::initializeEntity, cls);
            }
        }
```
## Possible Solution
As listed above, PR on its way...

## Steps to Reproduce (for bugs)
```java
    @Test
    public void test_subordinate_dictionarie_inherit_injector() {
        final Injector injector = 
             new Injector() {
                @Override
                public void inject(Object entity) {
                    throw new UnsupportedOperationException();
                }                
            };
        final QueryDictionaryDataStore ds1 = new QueryDictionaryDataStore();
        final MultiplexManager multiplexManager = new MultiplexManager(ds1);
        multiplexManager.populateEntityDictionary(
            new EntityDictionary(
                new HashMap<>(),
                injector
            )
        );
        assertEquals(
            ds1.getDictionary().getInjector(),
            injector
        );
    }
    
    private static class QueryDictionaryDataStore implements DataStore {
        private EntityDictionary dictionary;
        
        @Override
        public void populateEntityDictionary(final EntityDictionary dictionary) {
            this.dictionary = dictionary;
        }

        @Override
        public DataStoreTransaction beginTransaction() {
            throw new UnsupportedOperationException();
        }
        
        public EntityDictionary getDictionary() {
            return dictionary;
        }
        
    }
```

## Context
We are using ELIDE within a spring environment and wish to use multiplexed datastores along with field injection.

## Your Environment
* Elide version used: 4.6.4
* Environment name and version (Java 11): Springboot

### PR DESCRIPTION
**Pull request by gmckerrell**
Resolves #1379

## Description
make sure injector is propagated to sub-ordinate datastores in the context of the MultiplexManager

## Motivation and Context
See issue

## How Has This Been Tested?
unittest added

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -354,7 +354,6 @@
  * ConstraintViolationExceptions are propagated on flush (JPA Transaction)
  * Enable support for JPA @MapsId annotation on relationships so that client doesn't have
    to provide a dummy ID to make entity creation work.
```
Comment (gmckerrell) [inline changelog.md:356]:
```suggestion
   to provide a dummy ID to make entity creation work.
    * Cache all calls to getEntityBinding
```

File: changelog.md [hunk 2]
```diff
@@ -354,7 +354,7 @@
...
- * Cache all calls to getEntityBinding
+* Cache all calls to getEntityBinding
```
Comment (gmckerrell) [inline changelog.md:357]:
```suggestion
 * Cache all calls to getEntityBinding
```

File: elide-datastore/elide-datastore-multiplex/src/test/java/com/yahoo/elide/datastores/multiplex/MultiplexManagerTest.java [hunk 3]
```diff
@@ -126,4 +127,46 @@ public void partialCommitFailure() throws IOException {
             assertEquals(((FirstBean) list.get(0)).name, "name");
         }
     }
+    
+    @Test
+    public void test_subordinate_dictionary_inherits_injector() {
```
Comment (gmckerrell) [inline elide-datastore/elide-datastore-multiplex/src/test/java/com/yahoo/elide/datastores/multiplex/MultiplexManagerTest.java:132]:
```suggestion
    public void subordinateEntityDictionaryInheritsInjector() {
```

**Comment by aklish** [comment]
Thanks.  Great contribution!

**Pull request merged**